docs/guides/integration/jupyter.md

```diff
@@ -0,0 +1,146 @@
+# Using uv with Jupyter
+
+The [Jupyter](https://jupyter.org/) notebook is a popular tool for interactive computing, data
+analysis, and visualization. You can use Jupyter with uv in a few different ways, either to interact
+with a project, or as a standalone tool.
+
+## Using Jupyter within a project
+
+If you're working within a [project](../../concepts/projects.md), you can kick off a Jupyter server
+with access to the project's virtual environment via the following:
+
+```console
+$ uv run --with jupyter jupyter lab
+```
+
+Within the notebook, you can then import your project's modules as you would in any other `uv run`
+invocation. For example, if your project depends on `requests`, `import requests` will import
+`requests` from the project's virtual environment.
+
+If you're looking for read-only access to the project's virtual environment, then there's nothing
+more to it. However, if you need to install packages from within the notebook, there are a few extra
+details to consider.
+
+### Creating a kernel
+
+If you need to install packages from within the notebook, we recommend creating a dedicated kernel
+for your project. Kernels enable the Jupyter server to run in one environment, with individual
+notebooks running in their own, separate environments.
+
+In the context of uv, we can create a kernel for a project while installing Jupyter itself in an
+isolated environment, as in `uv run --with jupyter jupyter lab`. Creating a kernel for the project
+ensures that the notebook is hooked up to the correct environment, and that any packages installed
+from within the notebook are installed into the project's virtual environment.
+
+To create a kernel, you'll need to install `ipykernel` as a development dependency:
+
+```console
+$ uv add --dev ipykernel
+```
+
+Then, you can create the kernel for `project` with:
+
+```console
+$ uv run ipython kernel install --user --name=project
```

Comment:
Could be cool to have a command that emits the project name, so you could do `--name=$(uv ...)` as a one-liner that actually works in any project.